client/src/components/Form.tsx

```diff
@@ -17,120 +22,49 @@ type FormProps = {
-    oneNumber: (value: ValidationType) => /\d/.test(value as string) || 'צריכה להכיל מספר',
-    specialCharacter: (value: ValidationType) =>
-      /[`!@#$%^&*()_+\-=[\]{};':"\\|,.<>/?~]/.test(value as string) || 'צריכה להכיל סימן מיוחד'
-  }
-};
-
-const Errors = (name: string, errors: FieldErrors) => {
-  return (
-    <div className={classes['form-helper']}>
-      <ErrorMessage
-        errors={errors}
-        name={name}
-        render={({ messages }) =>
-          messages && Object.entries(messages).map(([type, message]) => <p key={type}>{message}</p>)
-        }
-      />
-    </div>
-  );
-};
-
-const Form = (props: FormProps) => {
-  const { onSubmit, name, email, password, submitLabel } = props;
+const Form = ({ onSubmit, name, email, password, submitLabel }: FormProps) => {
   const {
     register,
     handleSubmit,
     formState: { errors }
-  } = useForm<FormOptions>({
-    criteriaMode: 'all'
-  });
-  const submitHandler = React.useCallback(
-    (data: FormOptions) => {
-      const dataToSubmit: FormOptions = {};
-      if (name) {
-        dataToSubmit.name = data.name;
-      }
-      if (email) {
-        dataToSubmit.email = data.email;
-      }
-      if (password) {
-        dataToSubmit.password = data.password;
-      }
-      onSubmit(dataToSubmit);
-    },
-    [onSubmit, name, email, password]
-  );
-  return (
-    <>
-      <form onSubmit={handleSubmit(submitHandler)} className={classes.form}>
-        <div className={classes.formContainer}>
-          {name && (
-            <>
-              <FormField
-                inputType="text"
-                inputProps={register('name', nameValidations)}
-                label="שם"
-                inputClass={errors.name && classes['input-error']}
-                placeHolder="שם מלא"
-              />
-              {errors.name && Errors('name', errors)}
-            </>
-          )}
-          {email && (
-            <>
-              <FormField
-                inputType="text"
-                inputProps={register('email', emailValidations)}
-                label="אימייל"
-                inputClass={errors.email && classes['input-error']}
-              />
-              {errors.email && Errors('email', errors)}
-            </>
-          )}
-          {password && (
-            <>
-              <FormField
-                inputType="password"
-                inputProps={register('password', passwordValidations)}
-                label="סיסמא"
-                inputClass={errors.password && classes['input-error']}
-              />
-              {errors.password && Errors('password', errors)}
-            </>
-          )}
-        </div>
+  } = useForm<FormOptions>();
 
-        <button type="submit" className={classes['form-submit-btn']}>
-          {submitLabel}
-        </button>
+  return (
+    <div className="wrapper">
+      <form onSubmit={handleSubmit(onSubmit)} className="form">
```

Comment:
I personally think that making a single form for different pages could raise some issues while making changes or additions. For example, the way you choose email and password validators based on whether we want a name input. Or, when passing submit events we need to check whether some of the fields exist.

But if you don't think this is critical, leave it as is

Reply:
דיברתי עם יעל לגבי זה וזה מה שהיא ביקשה ממני לעשות